js/ojsxc.js

```diff
@@ -205,10 +205,12 @@
                 password: password
             },
             success: function(d) {
-                if (d.result === 'success' && d.data && d.data.serverType !== 'internal' && d.data.xmpp.url !== '' && d.data.xmpp.url !== null) {
+                if (d.result === 'success' && d.data && d.data.serverType !== 'internal' && d.data.xmpp.url !== '' && d.data.xmpp.url !== null) {                       jsxc.storage.setItem('serverType', d.data.serverType);
                     cb(d.data);
                 } else if (d.data && d.data.serverType === 'internal') {
```

Comment:
I think it make sense to use a "constant" or "enum" for all required server types.